libs/UGridNET/dll/src/UGridReader.cs

```diff
@@ -0,0 +1,208 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UGridNET.Extensions;
+using System.Runtime.InteropServices;
+
+namespace UGridNET
+{
+
+    public sealed class UGridReader : UGridBase
+    {
+        // requires using log4net;
+        //private static readonly ILog log = LogManager.GetLogger(typeof(UGridReader));
+
+        private bool disposed = false;
+
+        public UGridReader(string path) : base(path, 0)
+        {
+            try
+            {
+                ReadMesh1D();
+                ReadMesh2D();
+                ReadContacts();
+                ReadNetwork1D();
+            }
+            catch
+            {
+                // If initialization fails, dispose of any partially initialized resources
+                Dispose();
+                // Rethrow to inform the caller of the failure
+                throw;
+            }
+        }
+
+        ~UGridReader()
+        {
+            // Cleanup in case Dispose wasn't called
+            Dispose(false);
+        }
+
+        public new void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+            base.Dispose();
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (!disposed)
+            {
+                if (disposing)
+                {
+                    // clean up
+                }
+
+                disposed = true;
+
+                base.Dispose(disposing);
+            }
+        }
+
+
+        private void ReadMesh1D()
+        {
+            int count = 0;
+            Invoke(() => UGrid.ug_topology_get_count(fileID, TopologyType.Mesh1dTopology, ref count));
+            if (count > 0)
+            {
+                for (int i = 0; i < count; i++)
+                {
+                    Mesh1D mesh1D = new Mesh1D();
+                    Invoke(() => UGrid.ug_mesh1d_inq(fileID, i, mesh1D));
+                    mesh1D.Allocate();
+                    Invoke(() => UGrid.ug_mesh1d_get(fileID, i, mesh1D));
+                    mesh1DList.Add(mesh1D);
+                }
+            }
+        }
+
+        private void ReadMesh2D()
+        {
+            int count = 0;
+            Invoke(() => UGrid.ug_topology_get_count(fileID, TopologyType.Mesh2dTopology, ref count));
+            if (count > 0)
+            {
+                for (int i = 0; i < count; i++)
+                {
+                    Mesh2D mesh2D = new Mesh2D();
+                    Invoke(() => UGrid.ug_mesh2d_inq(fileID, i, mesh2D));
+                    mesh2D.Allocate();
+                    Invoke(() => UGrid.ug_mesh2d_get(fileID, i, mesh2D));
+                    mesh2DList.Add(mesh2D);
+
+                    //var name = mesh2D.name.CopyToArray<byte>(UGrid.name_long_length);
```

Comment:
in the final pr, consider removing commented code

Reply:
It is a private method of a class in a namespace. It is fine.